orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -691,33 +722,362 @@ Status TransformGraphForPipeline(
     ));
   }
 
-  auto fill_node_args = [&](const Graph& graph,
-                            const std::vector<const NodeArg*>& existed_node_args,
-                            std::vector<std::string>& new_node_arg_names,
-                            std::vector<const NodeArg*>& merged_node_args) {
-    merged_node_args.insert(merged_node_args.end(), existed_node_args.begin(), existed_node_args.end());
-    for (auto& name : new_node_arg_names) {
-      merged_node_args.push_back(graph.GetNodeArg(name));
+  ORT_RETURN_IF_ERROR(SetInputsOutputsAndResolve(graph, new_input_names, new_output_names));
+  return Status::OK();
+}
+
+void AddNewNodeArg(Graph& graph,
+                   const std::string& op_name,
+                   onnx::TensorProto_DataType type,
+                   std::vector<NodeArg*>& new_node_args,
+                   std::vector<std::string>& new_names) {
+  auto& event_id = CreateTypedNodeArg(graph, type, op_name);
+  new_names.push_back(event_id.Name());
+  new_node_args.push_back(&event_id);
+}
+
+template <typename T>
+void AddNewNodeArgAndInitializer(Graph& graph,
```

Comment:
// This function is used when you want to create a scalar constant in a graph. 
// It may create a NodeArg so that other Node can references its value.
// It also cerates an initializer to store its value.